zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -310,6 +332,43 @@ protected Object jobRun() throws Throwable {
     }
   }
 
+  private int getMaxParagraphOutput()
+  {
+    return replLoader.getZeppelinConfiguration().getInt(MAX_OUTPUT_KEY, 500000);
+  }
+
+
+  private String createTempFile(String message)
+  {
+    String configuredValue = replLoader.getZeppelinConfiguration()
+                                       .getString(OVERFLOW_OUTPUT_DIR, System.getenv("TMPDIR"));
+    File file;
+    try
+    {
+      Path path = Paths.get(configuredValue);
+      if (!Files.exists(path) || !Files.isDirectory(path))
+      {
+        throw new IllegalArgumentException("No directory named " + path.toString() +
+                                           " found. Please check your configuration");
+      }
+      file = path.resolve("paragraph-output-" + System.currentTimeMillis()).toFile();
+      file.deleteOnExit();
+      FileOutputStream out = new FileOutputStream(file);
+      out.write(message.getBytes(StandardCharsets.UTF_8));
```

Comment:
`Files.write(path, message.getBytes(StandardCharsets.UTF_8))` instead of using streams directly.